Chap_API_Publish.tex

```diff
@@ -74,9 +76,31 @@ \section{\code{PMIx_Publish}}
 The blocking form of this call will block until it has obtained confirmation from the datastore that the data is available for lookup. The \refarg{info} array can be released upon return from the blocking function call.
 
 Publishing duplicate keys is permitted provided they are published to different
-ranges. Duplicate keys being published on the same data range shall return the
+ranges. Custom ranges are consider different if they have different members.
+Duplicate keys being published on the same data range shall return the
 \refconst{PMIX_ERR_DUPLICATE_KEY} error.
 
+In some cases, implementations may be incapable of distinguishing which
```

Comment:
Note from 3Q 2022 ASC (Aug. 11, 2022):
 - Let's require `PMIX_DATA_TO_PUBLISH` as the only way to publish data.
   - This has some backward compatibility issues.
     - Open MPI and (maybe) MPICH would be impacted
   - We should add an error if `PMIX_DATA_TO_PUBLISH` is not provided in the publish call.
   - In v5, mark not using `PMIX_DATA_TO_PUBLISH` as deprecated.
 - Should we just create a new API with an additional required parameter for the data_to_publish (what to name it? `PMIx_Publish_data`?)
   - Then we just deprecate the old API instead of adding a new attribute to work around it.
```c
pmix_status_t PMIx_Publish_NEW(const pmix_info_t info_to_publish[], size_t ninfos,
                               const pmix_info_t info_directives[], size_t ndirs);
```

Reply:
Per IAWG on Aug. 15, 2022:
 * Add new `PMIx_Publish2` and `pmix_server_publish2_fn_t` with two sets of `pmix_info_t`s
 * Deprecate the `PMIx_Publish` API and corresponding `pmix_server_publish_fn_t` API.
 * Do not add the `PMIX_DATA_TO_PUBLISH` attribute. Instead, prefer the new API.
 * In the Revision appendix note, `PMIx_Publish` was deprecated and replaced by `PMIx_Publish2` (same with server API)
 * No longer an issue on the ordering of info data and directives (this approach avoids this whole issue)